.circleci/config.yml

```diff
@@ -385,6 +385,116 @@ jobs:
       - store_artifacts:
           path: packages/python/plotly/dist
 
+
+  build-doc:
+    resource_class: xlarge
+    docker:
+      # specify the version you desire here
+      # use `-browsers` prefix for selenium tests, e.g. `3.6.1-browsers`
+      - image: circleci/python:3.6.8-node
+
+
+        working_directory: ~/project
+
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+            - "d9:b0:da:5c:3e:8c:b6:b6:d5:b6:35:70:ce:fe:f5:b1"
+
+      - checkout
+
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "doc/requirements.txt" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+      - run:
+          name: install dependencies
+          command: |
+            cd doc
+            python3 -m venv venv
+            . venv/bin/activate
+            npm install electron@1.8.4
+            npm install orca
+            pip install -r requirements.txt
+            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+              pip uninstall plotly
+              cd ../packages/python/plotly
+              python3 setup.py install
+              cd ../../../doc      
+            fi 
+            echo 'export PATH="/home/circleci/project/doc/node_modules/.bin:$PATH"' >> $BASH_ENV
+            sudo apt-get install -y poppler-utils libxtst6 xvfb libgtk2.0-0 libgconf-2-4 libnss3 libasound2
+            cd ..
+
+      - save_cache:
+          paths:
+            - ./doc/venv
+            - ./doc/node_modules
+          key: v1-dependencies-{{ checksum "doc/requirements.txt" }}
+
+      - run:
+          name: make html
+          command: |
+            cd doc
+            . venv/bin/activate
+            echo ${mapbox_token} > python/.mapbox_token
+            make -kj8 || make -kj8
+            curl https://raw.githubusercontent.com/plotly/documentation/source-design-merge/front-matter-ci.py > front-matter-ci.py
+            curl https://raw.githubusercontent.com/plotly/documentation/source-design-merge/check-or-enforce-order.py > check-or-enforce-order.py
+            python front-matter-ci.py build/html
+            python check-or-enforce-order.py build/html
+            if [ "${CIRCLE_BRANCH}" == "doc-prod" ]; then
+              cd build/html
+              git init
+              git config user.name nicolaskruchten
+              git config user.email nicolas@plot.ly
+              git add *
+              git commit -m build
+              git push --force git@github.com:plotly/plotly.py.git doc-prod:built
```

Comment:
I don't think that `doc-prod:built` at the end will work here, I think this refers to the branch name of the repo you `init` a few lines higher, so the default branch is `master`

Reply:
oh yes thanks